### ISSUE TITLE
New Valkyrie Interceptor being assigned wrong number

### ISSUE THREAD

**Issue by ayrtondenner** [OpenApoc/OpenApoc#1404]
**Describe the bug**
When buying a new Valkyrie Interceptor, instead of using previous number plus 1, it is being assigned number 91

**To Reproduce**
1. Check number assigned to previous vehicle
2. Buy a new vehicle
3. Check wrong assigned number

**Expected behavior**
It should be assigned number of previous vehicle plus 1

**Savegame**
[test vehicle wrong number.zip](https://github.com/OpenApoc/OpenApoc/files/14968050/test.vehicle.wrong.number.zip)

**Screenshots**
![image](https://github.com/OpenApoc/OpenApoc/assets/13112588/9e82be81-4c34-4d77-89ce-d790a2c64af4)

**Comment by FilmBoy84** [comment]
See also #1312 

There is a reason why OpenApoc chose to randomly assign numbers - I cannot recall what it was

But there should certainly be an option to mimic OG behaviour as standard

**Comment by ayrtondenner** [comment]
Yep, I saw the comment made in that task, but I understand that they are different issues, so for me it makes sense to have this separated issue here.

**Comment by Kurtsley** [comment]
You original Valkyrie interceptor is part of the initial vehicles assigned to orgs when the game starts. These are always going to be the first vehicles assigned or purchased in the city so they will always be the lowest number. When you buy the next Valkyrie, it is after all other orgs have been assigned their own vehicles. This next Valkyrie will be assigned 91 because it is literally the 91st in the city. This is why #1312 needs to be implemented, to separate the XCOM vehicles from the rest of the city.

**Comment by Kurtsley** [comment]
If you buy more than one Valkyrie, they will correctly increment by one ex: 91, 92, 93.

**Comment by ayrtondenner** [comment]
Can the game create more vehicles? So after I buy Valkyrie 91, then the game creates more vehicles, so my next purchase will be assigned 101 for instance. Is that possible in current state?

**Comment by Kurtsley** [comment]
If you buy the vehicles when the game is paused, they will increment by one up to 9999 in theory although I'm not sure on the max number. The reason it jumps to 91 is due to other orgs initial vehicles being assigned. I don't know if orgs can replace lost vehicles, maybe someone else can answer that.

**Comment by XCOM-HUB** [comment]
This is not only an issue with the Valkyrie Interceptor, but all vehicles you purchase. As others have mentioned in issue #1312, it is because OpenApoc counts all other vehicles used by different organizations. I am in favor of a prefix attached to the name of vehicles not owned by X-COM so that the number of all purchased vehicles will always be incremented by one as it is in the original game. I am also in favor of an extra "bonus" feature that would make it possible to change the color of your vehicles so it would be easier to differentiate between X-Com vehicles and other organizations' vehicles. 🚗 As I have mentioned in #774

**Comment by FilmBoy84** [comment]
> I don't know if orgs can replace lost vehicles, maybe someone else can answer that.

They can in the OG and indeed this affects both marketplace availability and organisation finances IIRC

A viable strategy to bankrupt an org in the OG without destroying their buildings is to lure out craft with attacks on buildings that do not destroy the tile then down the craft the org will then attempt to purchase more (dependent on relations and finances) and you can whittle down their money over time
